### PR TITLE
kernel: enable pressure interface

### PR DESCRIPTION
**Pull request by graysky2**
PSI provides a canonical way to see resource pressure increases as they develop, with pressure metrics for three major resources: memory, CPU, and IO. PSI stats are like barometers that provide fair warning of impending resource shortages, enabling users to take more proactive, granular, and nuanced steps when resources start becoming scarce.

References:
* https://www.kernel.org/doc/html/latest/accounting/psi.html
* https://lwn.net/Articles/759781/

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/glinet_gl-mt6000
Run-tested: x86/64/AMD Cezanne, flogic/glinet_gl-mt6000

### CODE REVIEW

**Comment by graysky2** [comment]
@hauke @BKPepe wondering what you guys think about this?  I feel it is value-added.

**Comment by BKPepe** [comment]
Let's ping all Turris maintainers: @Zatharalex, @rmuzik, @miska  any thoughts about this? Hopefully, they will respond. 
I don't have any opinion on this one, as I won't be using it.

**Comment by hauke** [comment]
How big is the size increase of the kernel binary?

This is only needed to debug problems. If it increases the size of the kernel binary it would be better suited as KERNEL_PSI for config/Config-kernel.in.

**Comment by graysky2** [comment]
@hauke - what is the kernel binary?  I see only two text files:
```
# opkg files kernel
Package kernel (6.6.22~c48493ba0f5c21ae91a9df7c2d5cacfa-r1) is installed on root and has the following files:
/lib/modules/6.6.22/modules.builtin
/lib/modules/6.6.22/modules.builtin.modinfo
```

**Comment by graysky2** [comment]
> How big is the size increase of the kernel binary?
@hauke - can you comment on how I can see the kernel binary size?

**Comment by graysky2** [comment]
>How big is the size increase of the kernel binary?

For x86/64 the difference is approx 8kB

Without this PR:
```
% ls -l ./targets/x86/64/openwrt-x86-64-generic-kernel.bin 
-rw-r--r-- 1 facade users 6116352 Nov 23 11:05 ./targets/x86/64/openwrt-x86-64-generic-kernel.bin
```
With this PR:
```
% ls -l ./targets/x86/64/openwrt-x86-64-generic-kernel.bin
-rw-r--r-- 1 facade users 6124544 Nov 23 11:10 ./targets/x86/64/openwrt-x86-64-generic-kernel.bin
```

**Comment by hauke** [comment]
I do not think we need this activated by default but it is helpful to be able to easily activate it. I would prefer this solution instead:
```
diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index 0fa039db40..91678cf2a6 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -333,6 +333,25 @@ if KERNEL_TASKSTATS
 
 endif
 
+config KERNEL_PSI
+	bool "Compile the kernel with pressure stall information tracking"
+	help
+	  Collect metrics that indicate how overcommitted the CPU, memory,
+	  and IO capacity are in the system.
+
+	  If you say Y here, the kernel will create /proc/pressure/ with the
+	  pressure statistics files cpu, memory, and io. These will indicate
+	  the share of walltime in which some or all tasks in the system are
+	  delayed due to contention of the respective resource.
+
+	  In kernels with cgroup support, cgroups (cgroup2 only) will
+	  have cpu.pressure, memory.pressure, and io.pressure files,
+	  which aggregate pressure stalls for the grouped tasks only.
+
+	  For more details see Documentation/accounting/psi.rst.
+
+	  Say N if unsure.
+
 config KERNEL_KALLSYMS
 	bool "Compile the kernel with symbol table information"
 	default y if !SMALL_FLASH
```

**Comment by graysky2** [comment]
Since this is your work, please add your own line to the commit message, thanks!

File: config/Config-kernel.in [hunk 1]
```diff
@@ -333,6 +333,30 @@ if KERNEL_TASKSTATS
...
+
+	  In kernels with cgroup support, cgroups (cgroup2 only) will
+	  have cpu.pressure, memory.pressure, and io.pressure files,
+	  which aggregate pressure stalls for the grouped tasks only.
+
+	  For more details see Documentation/accounting/psi.rst.
+
+	  Say N if unsure.
+
+config PSI_DEFAULT_DISABLED
```
Comment (hauke) [inline config/Config-kernel.in:355]: Kernel config options have to start with `KERNEL_` in this file. 

```suggestion
config KERNEL_PSI_DEFAULT_DISABLED
```

Comment (graysky2) [reply config/Config-kernel.in]: Done

**Pull request merged**